### ISSUE TITLE
Replace turtle boss with something based on army

### ISSUE THREAD

**Issue by Quintus** [Secretchronicles/TSC#77]
The turtle boss still resembles way to much the original SMB. It has to be replaced with something based on the new army graphics.

See als [this forum thread](http://smc.quintilianus.eu/viewtopic.php?id=33).

Vale,
Quintus

**Comment by Bugsbane** [comment]
I will finish this, but it's fairly low priority compared to most other assets as it's something fairly few people will see, compared to almost every other asset currently needed.

That said, it's something that will be fund to finish, so I'm sure I'll likely get drawn into (pardon the pun) earlier than I should.

**Comment by Quintus** [comment]
Can’t wait to see this guy live and in colour :-).

Vale,
Quintus

**Comment by Bugsbane** [comment]
> Can’t wait to see this guy live and in colour :-)

Me, too. He'll be fun to finish. :)

**Comment by datahead8888** [comment]
Besides the fact that this boss will be really cool with Bugsbane's concept shown above, it would be great to eliminate the final reference to Koopa Troopas from the Mario games.

**Comment by Bugsbane** [comment]
True.

**Comment by Quintus** [comment]
I suggest to remove the turtle boss from the game for now. We can add a revamped version after 2.0.0 when @Bugsbane has time to finish his terrificly cool guy :-)

Valete,
Quintus

**Comment by datahead8888** [comment]
> I suggest to remove the turtle boss from the game for now.

As discussed in IRC, I feel a bit hesitant to remove the turtle boss because this is the removal of a major feature (bosses are a big deal in a platformer, and we only have 2).  I understand Quintus's point, since this is a clear-cut derivative work of Nintendo assets, just like our Mario based music is.  Nintendo could call this copyright infringement.

Since removing the turtle boss was just suggested yesterday, I'd at least like to discuss this some more before removing him.  I know everyone wants to get the release done but think it's good to have more conversation on a major change like this.  Release 2.0 is probably the only release where first impressions are going to matter to any extent.

**Comment by datahead8888** [comment]
@SonOfBower - you had strongly objected to removing the hedge because it impacts users' levels.  We are talking about completely removing the turtle boss for the same reason.  If you have any input to this decision, we're always interested in hearing feedback.

**Comment by datahead8888** [comment]
@SonOfBowser - you had strongly objected to removing the hedge because it impacts users' levels.  We are talking about completely removing the turtle boss for the same reason.  If you have any input to this decision, we're always interested in hearing feedback.

**Comment by SonOfBowser** [comment]
The issue was more with leaving bundled levels that include removed objects. This seems like an even bigger issue for something like this, where the boss acts as an end of level, so levels including it probably won't have defined exits. If someone wants to remove it and remove/ update all levels that include it, I see no problem there (other than the already discussed lack of bosses).

**Comment by Quintus** [comment]
There’s exactly one level in the official campaign that uses the turtle boss in question here, so that shouldn’t be a problem. I’m not sure about the other levels in the game, but I don’t think I’ve ever seen any using the turtle boss.

Vale,
Quintus

**Comment by datahead8888** [comment]
> There’s exactly one level in the official campaign that uses the turtle boss in question here, so that shouldn’t be a problem. I’m not sure about the other levels in the game, but I don’t think I’ve ever seen any using the turtle boss.

The other consideration is user submitted levels that are not packaged with the game.

I'm more concerned with the number of bosses available in the game, though.

**Comment by datahead8888** [comment]
@Bugsbane - if you were to do a "faster" version of the Turtle Boss replacement (ie not focusing as hard on making him look good), how long would this task take?  Might it be possible to just rescale some of the existing army and army shell graphics into army boss graphics?  I realize there probably is not a 1:1 correspondence, which would require you to fill in animation gaps.  A more detailed version of the army boss that looks really good could be done later.

Sydney offered to help with the 3-Up moon replacement, which would let you look at a simpler Army Boss if you think it is feasible.  You're always welcome to get involved in the moon replacement if you feel it best, though.

If the turtle boss cannot be replaced soon, I agree with Quintus the feature must be removed from release 2.0 completely.  We cannot delay the release any further.  The moon graphic likewise has to be replaced, but it's a little harder for us just to remove that feature from the game.

**Comment by Bugsbane** [comment]
Hey guys,
I might be able to do something very simplified next weekend (March 7) just with very few frames of animation. I'd like to see what art asset blockers there are for 2.0.0 release first though as I still feel that while cool, the boss is something few people will actually see versus many other assets that are seen early and often by many people (eg many enemies).

Basically, I want to make sure that if I put aside a couple of hours to work on assets for TSC that I'm working on what will have the most impact to the largest number of players.

This isn't about whether this issue is important. It's about which art asset issue(s) is/are the *most* important.

**Comment by datahead8888** [comment]
@Bugsbane - thanks for offering to help!  Unfortunately, things have changed a little, and we probably need your help with the moon replacement (we were looking at a pineapple based on Sydney's suggestion but are open to more discussion).  As much as I'd love to keep the turtle boss in release 2.0 (replaced), it is probably harder to remove the moons than the turtle boss.

> How up to date are the issue tags? Can I just do a search for something like tag: art +tag:blocker and get a list that I can trust is comprehensive and up to date?

I updated the turtle boss and moon tasks with Blocker status for you.

We probably should come to a decision on #298 if possible before the release.  I may email Luiji to see if he can offer an outside perspective and see if we can avoid voting.  I would really like to abstain from taking a position on that asset myself.

We still need to commit the star replacement (#267) to git, but we agreed the current graphics are ready for the release.

**Comment by Quintus** [comment]
As per [decision of the General Discussion](http://www.secretchronicles.de/en/news/2015/03/22/results-of-first-general-discussion/) I have disabled the turtle boss for now and replaced the level containing the turtle boss with `mountain_trials`. When we have a turtle boss replacement available, things can be undoed.

I remove this from the 2.0.0 milestone now.

Valete,
Quintus

**Comment by datahead8888** [comment]
In the IRC meeting I had suggested replacing the current turtle boss level with the furball boss level and the current furball boss level with Mountain Trials.  This might yield a progressive difficulty and would put Mountain Trials near the end.  Would this sound like a good idea in the release?  I know it is difficult to remember exact suggestions after such a long meeting.

**Comment by Quintus** [comment]
> In the IRC meeting I had suggested replacing the current turtle boss level with the furball boss level and the current furball boss level with Mountain Trials.

Oops, then I misunderstood you. I replaced the turtle boss level with mountain trials :-). I’m fine with either, feel free to edit the worlds just as you like.

Vale,
Quintus

**Comment by datahead8888** [comment]
I may not be able to get to it for a bit if I try to do it, unless anyone else is interested in making the change.

**Comment by datahead8888** [comment]
> I may not be able to get to it for a bit if I try to do it, unless anyone else is interested in making the change.

It will have to stay this way in release 2.0.  I cannot make any more changes to TSC until about May.  I also will try to be in IRC and the forums less.

**Comment by datahead8888** [comment]
> Oops, then I misunderstood you. I replaced the turtle boss level with mountain trials :-). I’m fine with either, feel free to edit the worlds just as you like.

@Quintus - my research indicates the following:
* The only level other than mountain_trials.tsclvl that uses the furball boss is sauer2_7_boss.tsclvl
* sauer2_7_boss.tsclvl is not currently used in the campaign.  It is neither referenced in any level xml files nor any world xml files.

Does this match your observations?  I could have sworn sauer2_7_boss.tsclvl was used before in the campaign.

If this is correct, the progression of difficulty in boss related levels is fine, since only one level has a boss at all.  It would be good to consider moving Mountain Trials to World 4 still, but at least we don't go from an easier boss level to a harder boss level from one world to the next.

**Comment by Quintus** [comment]
sauer2_7_boss should be the level referenced from the last level in world 4 as a sublevel. I have to check that.

Vale,
Quintus

**Comment by Quintus** [comment]
Wat. You’re right. The red exit door in sauer2_7 is a regular level exit now. @sauer2, did you change that? If so, shouldn’t we change it back?

Vale,
Quintus

**Comment by datahead8888** [comment]
> Wat. You’re right. The red exit door in sauer2_7 is a regular level exit now. @sauer2, did you change that? If so, shouldn’t we change it back?

I have not made any changes.  If you end up editing this stuff in the world editor, could you quickly do the boss level swap we've been talking about?  The user would see sauer_2_7_boss first in the game and then Mountain_Trials.  I'd really appreciate it :)

**Comment by Quintus** [comment]
It’s not that easy. sauer2_7 is a fairly hard level, you can’t just move it into world 1. sauer2_7_boss without sauer2_7 doesn’t make sense either. I’d like to have @sauer2’s opinion of this.

Vale,
Quintus

**Comment by datahead8888** [comment]
If so, we definitely need to discuss before proceeding.  Sauer2's opinion will be invaluable.

**Comment by sauer2** [comment]
> If so, we definitely need to discuss before proceeding. Sauer2's 
> opinion will be invaluable.
>

I didn't change the exit door; at least not intentionally, it may have 
happened while editing, sorry.

However I don't think those two levels should be warp-connected as the 
levels they are currently.

Not so much because of sauer2_7, but the boss level, which is pretty 
hard due to the 15 lifes of the boss. Replaying sauer2_7 just to beat 
the boss would be pretty tedious.

In any way, a downgrade to 10 or even 7 lifes might be appropriate.

**Comment by datahead8888** [comment]
> In any way, a downgrade to 10 or even 7 lifes might be appropriate.

Is the furball boss in the Mountain Trials level easier?  If so, sauer2_7_boss actually may be good in that the game becomes progressively more difficult.  If not, we may consider swapping the order of the two levels as described above.

**Comment by sauer2** [comment]
> Is the furball boss in the Mountain Trials level easier? If so, 
> sauer2_7_boss actually may be good in that the game becomes 
> progressively more difficult. If not, we may consider swapping the 
> order of the two levels as described above.
>
Yes, the furball in Mountain Trials has only 3 lifes.

**Comment by datahead8888** [comment]
In my opinion, if a level accidentally got removed that we wanted in the game, this is a bug to be fixed for the release candidate.  Thus, even if we freeze everything else, this still should be fixed before releasing release 2.0, I think.

**Comment by Quintus** [comment]
So we just introduce another level behind sauer2_7 then? I’d be fine with this.

Valete,
Quintus

**Comment by datahead8888** [comment]
@Bugsbane - do you think you might get a chance to work on this?  I know it will take a while, though maybe it can be done a little at a time :)

**Comment by Bugsbane** [comment]
OK, update time. I took a few hours to go through and trace, colour and do the linework for the first image, which is actually 90% of the work for image 2 and 3 as well. Basically you get everything looking just right on one, and then the other two are largely just grabbing objects, moving and rotating them.

Still do do:
* Adjust final line widths and colours
* Add shadows and highlights
* The 4 "in shell" frames (although 2 are just horizontally flipped copies of the other 2)
* Misc. details (spots, hairs, wrinkles, fixing inkscape glitches etc)

![walk_1](https://cloud.githubusercontent.com/assets/401733/12196187/b8ac93d6-b5ca-11e5-9bc0-087d2970cc31.png)

**Comment by datahead8888** [comment]
Wow, I can't wait to see the final product, in the game!

**Comment by Quintus** [comment]
AWESOME

**Comment by Quintus** [comment]
Btw. Bugsbane, if you need more frames to make it look smooth, we will make that possible by changing the source code. Don’t force yourself into the exact number of frames the old turtle boss had.

Vale,
Quintus

**Comment by Bugsbane** [comment]
What I'll do will be to replace the original frames first to get him in ASAP, afterwards if I have some time I'll do some inbetween frames. The rest of it shouldn't take much longer, as the toughest work is already done here.

**Comment by Bugsbane** [comment]
Cleaned up a bit. There's always going to be 100 things I'd like to fix, but I think he's just about ready for animation.

![walk_1](https://cloud.githubusercontent.com/assets/401733/12210344/38b2b5c2-b627-11e5-964d-fc63903dd5f6.png)

**Comment by Quintus** [comment]
There seem to be some pixel errors in its left feet’s outline. I know that they’re probably invisible when scaled to game size, but you might still take a look at it.

Thank you for your efforts, Bugsbane! What would TSC be without you...

Vale,
Quintus

**Comment by Bugsbane** [comment]
Yeah, I'm aware of those pixel errors (and a number of others). They seem to come from some bug in Inkscape's power outline path effect. I'm working with the team there to try and get it resolved. Apparently not many people use variable line width as much as I like to...

**Comment by Bugsbane** [comment]
Hey, I just had an idea that I think would let me turn out the needed images quicker and look better too, but it would require some code changes, so I wanted to ask what you guys thought.

Specifically, I'm thinking that rather than have him spinning around horizontally using 4 frames of animation as it does now, that it would actually look better and match an armadillo more, if I provided a single frame of him curled up in a ball, and we rotated that as he rolled (instead of spinning horizontally), much like we do with Spika.

* This way his animation can be frame perfect with just a single image, instead of needing 4 images that are only updated every 8 frames or so.
* He looks more like an armadillo (rolling, instead of spinning on his stomach).
* It makes it a bit more different than the Mario games, where the bosses typically rotate horizontally, based on frame animation.
* The needed frames could be delivered faster as there's 3 less needed.

We could even use some kind of circular saw sound effect when he was rolling, possibly like a slightly modified, looping version of [this](http://freesound.org/people/kwahmah_02/sounds/250054/). (I could take care of creating the sound).

All we'd need is basically just code to rotate the sprite when the boss moves, clockwise when he moves right and anticlockwise when he moves left.

**Comment by Quintus** [comment]
Sounds like a nice idea. Given we have done that with the spika, it should be possible to do that here to. If we had ported everything over to a complete SFML-based architecture already I would even volunteer to do that myself since [it is really easy to do there](http://www.sfml-dev.org/documentation/2.3.2/classsf_1_1Sprite.php#af8a5ffddc0d93f238fee3bf8efe1ebda), but currently we need raw OpenGL code to do that (I learned from various tickets one uses `glRotatef()` for that meanwhile, though). I still think this is a nice idea, though.

Vale,
Quintus

**Comment by Quintus** [comment]
Well, it appears the call is wrapped already in `cSprite::Add_Rotation_Z()`. It is possible to do this for me also.

Vale,
Quintus

**Comment by datahead8888** [comment]
I'd be interested in helping if needed, since I'd really like to see this in the next release.  I should try to stay focused on the SFML task, however, unless you need help with something in OpenGL.

**Comment by datahead8888** [comment]
@Bugsbane, have you had a chance to work on this recently?  I would love to see it in the next release if we can manage it :)

**Comment by Bugsbane** [comment]
Yes. I'm just trying to learn a new tool (Synfig) which would let me add new animations far more quickly and with much smoother frames of animation. The rigging is proving a bit tricky to learn my first time though, which is slowing things down. If it was desperate, I could still do the frames manually. It would just waste some time off I then also went back and set things up in Synfig later.

**Comment by datahead8888** [comment]
Thanks, Bugsbane.  This is a very exciting change to the game :)

**Comment by datahead8888** [comment]
@Bugsbane - have you had a chance to work on this recently?  We are talking about doing a release semi-soon, and this would be a very cool addition for it.

**Comment by datahead8888** [comment]
If we aren't sure when a new version of the Army Boss graphics will be done, we could consider this:
* Reprogram the C++ code for the former Turtle boss in order to use the same number of images as the regular Army enemies
* Replace the sprites for the reprogrammed Turtle boss with the sprites for the regular Army enemy.  Scale them to be large enough for the size.  Because we are scaling vector SVG graphics, there should not be a quality loss.
* Put the reprogrammed Army boss back in the game as a temporary solution.  When new graphics are created for the Army boss, we can restore the old C++ code (tweaking it for additional frames the Turtle boss never had in SMC if needed).

We've discussed doing the same thing for the Larry boss in #100.  This would allow us to have two bosses in the 2.X series again and could also offer a way to have a third boss.  Artists can then polish the look and feel for both bosses when they get a chance.

**Comment by datahead8888** [comment]
To be clear, when I suggest reprogramming the C++ code for the Turtle/Army boss, I mean reducing the number of moves it uses and giving it different behavior.  This is what allows less frames to be used.

**Comment by Quintus** [comment]
I think these suggestions are good and we should do it like that. Do you
intend to work on this after your joystick work is finished?

Vale,
Quintus

-- 
Blog: http://www.guelkerdev.de
PGP/GPG ID: F1D8799FBCC8BC4F

**Comment by datahead8888** [comment]
> Do you intend to work on this after your joystick work is finished?

I need to make some movement on some other projects besides TSC, so both further joystick fixes (if there are other issues) and features aren't known for sure.

**Comment by Quintus** [comment]
@Bugsbane ?

**Comment by datahead8888** [comment]
Adding the feature label to reflect the reprogramming that we are looking at doing

**Comment by datahead8888** [comment]
I'm not really sure this will move fast at all for a while, but I would be interested in looking into this feature.
Also, Quintus had thrown out the idea of a feature freeze after finishing the alternate exits feature.  I propose that this army boss feature and anything required for changing theme songs be exempt from a feature freeze.

**Comment by datahead8888** [comment]
The basic idea is that we removed a boss in release 2.0, so getting the boss back has a bit more priority than the average feature.

**Comment by datahead8888** [comment]
As for the graphics revisions for the new Army boss, I by no means mean to discourage Bugsbane from working on them -- they just should probably go under a new ticket.  This ticket will represent programming a new version of the boss making use of resized Army and Army shell sprites.

**Comment by Quintus** [comment]
Instead of "exempting" this from the feature freeze, just officially
include it in the list of features for 2.1.0. However, rather than
having this ticket represent the programming task and have the graphics
go into a new one, I want to suggest the inverse: Let this ticket remain
the graphics task and open a new one for the programming task. This
allows the entire discussion on the graphics including drafts of the new
boss to remain here, and allows Bugsbane to easier catch up when he
later returns as he is subscribed to this ticket.

The 2.1.0 milestone would then be set for the new ticket.

-- 
Blog: https://www.guelkerdev.de
PGP/GPG ID: F1D8799FBCC8BC4F

**Comment by datahead8888** [comment]
> However, rather than having this ticket represent the programming task and have the graphics go into a new one, I want to suggest the inverse: Let this ticket remain the graphics task and open a new one for the programming task.

This is now done with ticket #606, set to the 2.1 milestone.  I have reassigned this ticket to Bugsbane.

I'd like to put milestone 2.1 for the Larry boss possibly, though that one may or may not happen.  I'd be interested in keeping the door open for that possibility for now if we can, but I'm open to other views.  The only other thing I think should have the 2.1 milestone other than alternate exits is any code changes if needed to reintroduce removed theme songs, supposing we had new ones and that code was required.

Aside from that I agree with a feature freeze.
